### ISSUE TITLE
Selectstmt with redundant parentheses is not support.

### ISSUE THREAD

**Issue by wshwsh12** [pingcap/tidb#12610]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
Open a client, and input the following SQL:
```
mysql> create table t(a int);
Query OK, 0 rows affected (0.01 sec)

mysql> ((select * from t));
Empty set (0.00 sec)

mysql> (((((select * from t)))));
Empty set (0.00 sec)
```

2. What did you expect to see?

Above.

3. What did you see instead?
```
tidb> create table t(a int);
Query OK, 0 rows affected (0.00 sec)

tidb> ((select * from t));
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 19 near ")"
tidb> (((((select * from t)))));
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 3 near "(((select * from t)))))"
```


4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
```

tidb> select tidb_version();
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                                                               |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v4.0.0-alpha-455-g2354fad90
Git Commit Hash: 2354fad905d113e29612ae1b941d8d17ca50e6c1
Git Branch: master
UTC Build Time: 2019-10-10 05:41:59
GoVersion: go version go1.13 darwin/amd64
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)
```

**Comment by nullnotnil** [comment]
Confirming that this is still an issue in master:
```sql
drop table if exists t;
create table t(a int);
((select * from t));
(((((select * from t)))));

..

mysql> ((select * from t));
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 19 near ")" 
mysql> (((((select * from t)))));
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 3 near "(((select * from t)))))" 
mysql> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v4.0.0-beta.2-771-gca41972fb
Edition: Community
Git Commit Hash: ca41972fbac068c8a5de107d9075f09ac68842ac
Git Branch: master
UTC Build Time: 2020-07-14 02:41:21
GoVersion: go1.13
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
1 row in set (0.00 sec)

```

**Comment by wjhuang2016** [comment]
Close this issue since it is fixed by pingcap/parser#960

And after the discussion with @lzmhhh123 , we decide not to cherry-pick to 4.0/3.0 because:

The work is non-trivial.
The syntax is a corner case and does not lead to a serious problem.
